apps/angular/5-crud-application/src/app/app.component.ts

```diff
@@ -1,51 +1,29 @@
 import { CommonModule } from '@angular/common';
-import { HttpClient } from '@angular/common/http';
-import { Component, OnInit } from '@angular/core';
-import { randText } from '@ngneat/falso';
-
+import { Component, signal } from '@angular/core';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { ToDoComponent } from '../components/to-do/to-do.component';
+import { TodoDetails } from '../model/user';
+import { CrudSpecialService } from '../services/crud-special.service';
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, MatProgressSpinnerModule, ToDoComponent],
   selector: 'app-root',
-  template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
-    </div>
-  `,
+  templateUrl: './app.component.html',
   styles: [],
 })
-export class AppComponent implements OnInit {
-  todos!: any[];
+export class AppComponent {
+  public loading = true;
+
+  constructor(private todoService: CrudSpecialService) {}
 
-  constructor(private http: HttpClient) {}
+  todos = this.todoService.allTodos;
+  todoList!: TodoDetails[];
 
-  ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        this.todos = todos;
-      });
+  updateTodo() {
+    this.todoList = this.todos();
   }
 
-  update(todo: any) {
-    this.http
-      .put<any>(
-        `https://jsonplaceholder.typicode.com/todos/${todo.id}`,
-        JSON.stringify({
-          todo: todo.id,
-          title: randText(),
-          body: todo.body,
-          userId: todo.userId,
-        }),
-        {
-          headers: {
-            'Content-type': 'application/json; charset=UTF-8',
-          },
-        },
-      )
-      .subscribe((todoUpdated: any) => {
-        this.todos[todoUpdated.id - 1] = todoUpdated;
-      });
+  deleteTodo($event: TodoDetails) {
+    this.todos = signal([...this.todos().filter((g) => g.id !== $event.id)]);
```

Comment:
if you want to update your signal, use update or set method. Don't recreate a signal